src/main/java/com/google/cloud/teleport/io/WindowedFilenamePolicy.java

```diff
@@ -137,22 +156,37 @@ public ResourceId unwindowedFilename(
   /**
    * Resolves any date variables which exist in the output directory path. This allows for the
    * dynamically changing of the output location based on the window end time.
+   * A custom pattern can be used, if provided.
    *
    * @return The new output directory with all variables resolved.
    */
   private ResourceId resolveWithDateTemplates(
       ValueProvider<String> outputDirectoryStr, BoundedWindow window) {
     ResourceId outputDirectory = FileSystems.matchNewResource(outputDirectoryStr.get(), true);
-
+  
     if (window instanceof IntervalWindow) {
       IntervalWindow intervalWindow = (IntervalWindow) window;
       DateTime time = intervalWindow.end().toDateTime();
       String outputPath = outputDirectory.toString();
-      outputPath = outputPath.replace("YYYY", YEAR.print(time));
-      outputPath = outputPath.replace("MM", MONTH.print(time));
-      outputPath = outputPath.replace("DD", DAY.print(time));
-      outputPath = outputPath.replace("HH", HOUR.print(time));
-      outputPath = outputPath.replace("mm", MINUTE.print(time));
+      String customDateTimePattern = pattern.get();
```

Comment:
Can this never be null?

Reply:
I was setting an empty String as the default value for customDateTimePattern. Added a check now to avoid the NPE.